### PR TITLE
[aot] C_API behavior test

### PR DESCRIPTION
**Pull request by damnkk**
Issue: #

### Brief Summary
Added run out of memory test case.

### CODE REVIEW

File: taichi/rhi/vulkan/vulkan_api.cpp
Comment (PENGUINLIONG) [inline taichi/rhi/vulkan/vulkan_api.cpp:518]: // FIXME: (damnkk) Should be removed when RHI error codes are ready

**Review by PENGUINLIONG — CHANGES_REQUESTED** [review]

**Comment by PENGUINLIONG** [comment]
There is an on-going rafactorization to add error codes to all RHI interfaces. We don't have it for now so I'm not sure if it's a proper way to report OOM errors for the release. Or we just wait for a future commit of OOM RHI error to be merged? wdyt @bobcao3 @ailzhang

File: c_api/tests/c_api_behavior_test.cpp
Comment (ailzhang) [inline c_api/tests/c_api_behavior_test.cpp:122]: nit: early return after warning?

File: c_api/tests/c_api_behavior_test.cpp
Comment (ailzhang) [inline c_api/tests/c_api_behavior_test.cpp:125]: nit: better split these to 5 smaller / separate tests to keep tests minimal and isolated ;)

**Comment by ailzhang** [comment]
IMHO it doesn't have to be blocked by the missing OOM error code in rhi refactorization as long as we add a FIXME and fix it right after the merge. Adding these tests will also help make sure the refactorization goes well.

**Review by PENGUINLIONG — APPROVED** [review]
lgtm then. thank you!

**Pull request merged**
